apps/meteor/app/apps/server/bridges/rooms.ts

```diff
@@ -102,6 +102,55 @@ export class AppRoomBridge extends RoomBridge {
 		return this.orch.getConverters()?.get('users').convertById(room.u._id);
 	}
 
+	protected async getMessages(
+		roomId: string,
+		options: {
+			limit: number;
+			skip?: number;
+			sort?: Record<string, 1 | -1>;
+		},
+		appId: string,
+	): Promise<IMessageRaw[]> {
+		this.orch.debugLog(`The App ${appId} is getting the messages of the room: "${roomId}"`);
+
+		let { limit, skip = 0, sort } = options;
+
+		if (!Number.isFinite(limit) || limit < 1) {
+			limit = 100;
```

Comment:
if the numbers are invalid why not throw some error instead?

Reply:
We can but we didn't.

There is going to be a default error from TS and compiler when the Numbers are invalid unless someone asserts something like `limit: "string" as unknown as Number` which was intentional. So taking the approach of default behavior and returning the max limit `100`.

Reply:
Forget about the compiler, since the app developer still can just use JS. Why we can't throw a runtime error here? Do we have a policy of being merciful at logic/calculation errors at production? If that's so, at least we should log warnings for the developer at development mode (if there is one). Otherwise, throwing a runtime error is a warranty for future changes not being blocked by this kind of leniency.

Reply:
Agree with Gazzo and Tasso. Not to clutter the comment section, but adding a different perspective - we at SRE have had tons of problems due to no-logs. One, had literally yesterday (not core related). There needs to be a 1:1 correlation between what we see and what we get, exceptions with verbosity for easier detection. IOW we should be able to figure out some path from reading logs, get some idea.

A log, or throw error is good. Throw error might be better as then we just don't compute anything for wrong code/calculation.

Reply:
Thank you for the review.

No, we don't have a specific policy of being merciful. 😄 

The `limit` is optional on the Apps Engine end. You can just call `getMessages(roomed),`, and it will return "100" messages. So even if you call with a `NaN` we just return the default.

I think logging the events would be a nice idea.

Reply:
@debdutdeb @tassoevan those are good points. There were no discussions or hard stances on not throwing an error at this point, I think @Dnouv just didn't do it this way.

Reply:
Removed the checks, since we make sure on Apps Engine side to return a valid number. I will note the decision on ADR as well. Thanks!

Reply:
This isn't resolved yet, moving the same code to apps engine makes no difference

https://github.com/RocketChat/Rocket.Chat.Apps-engine/pull/770/files#diff-b4205e5b20daca4b90bb2ad8f137fce587773eea26bf35fb7df28196a327ea52R33-R46

it's still neither logging nor throwing error. What difference is it making?

Reply:
We log what `options` are passed. https://github.com/RocketChat/Rocket.Chat/blob/f35fa9a037fc4698f419eb4d4e6a43ef50089698/apps/meteor/app/apps/server/bridges/rooms.ts#L114

> What difference is it making?

On the bridge code (rc side) we know the `limit` will always be a valid number. We were limiting the value to `100` on Apps Engine's accessor if no `limit` is passed from `user` so extended it further to validate a number as well.

Reply:
I previously used the way to go with default value instead of throwing an error because we do similar for API calls: https://github.com/RocketChat/Rocket.Chat/blob/fd2aee8f8839b02bf14decd44911a5dc9766e7f6/apps/meteor/app/api/server/helpers/getPaginationItems.ts#L6 

In my opinion, it provides a better user experience.

Reply:
We throw an error now, 
```
if (typeof options.limit !== 'undefined' && (!Number.isFinite(options.limit) || options.limit > 100)) {
            throw new Error(`Invalid limit provided. Expected number <= 100, got ${options.limit}`);
        }
```